src/client/tile.cpp

```diff
@@ -1765,6 +1765,51 @@ bool TextureSource::generateImagePart(std::string part_of_name,
 				baseimg->setPixel(x,y,c);
 			}
 		}
+		/*
+			[invert:mode
+			Inverts the given channels of the base image.
+			Mode may contain the characters "r", "g", "b", "a".
+			Only the channels that are mentioned in the mode string
+			will be inverted.
+		*/
+		else if (str_starts_with(part_of_name, "[invert:")) {
+			if (baseimg == NULL) {
+				errorstream << "generateImagePart(): baseimg == NULL "
+						<< "for part_of_name=\"" << part_of_name
+						<< "\", cancelling." << std::endl;
+				return false;
+			}
+
+			Strfnd sf(part_of_name);
+			sf.next(":");
+
+			std::string mode = sf.next("");
+			bool r = mode.find("r") != std::string::npos;
+			bool g = mode.find("g") != std::string::npos;
+			bool b = mode.find("b") != std::string::npos;
+			bool a = mode.find("a") != std::string::npos;
+
+			core::dimension2d<u32> dim = baseimg->getDimension();
+
+			for (u32 y = 0; y < dim.Height; y++)
+			for (u32 x = 0; x < dim.Width; x++)
+			{
+				video::SColor c = baseimg->getPixel(x,y);
+				if (r) {
+					c.setRed(255 - c.getRed());
+				}
+				if (g) {
+					c.setGreen(255- c.getGreen());
```

Comment:
Space missing before `-`. The brackets (`{`, `}`) can be left away for single instructions, like it's the case here.